src/main.js

```diff
@@ -55,8 +55,17 @@ if (window.location.search.indexOf("testEnvironment") > -1) {
      * navigator.language in ExtensionLoader (when making require contexts for each
      * extension).
      */
+    var locale;
+    var regex = /locale=([^&]+)&?/.exec(window.location.search);
+    if(window.location.search && regex !== null) {
```

Comment:
Are you concerned that `window.location.search` won't exist? That would be the reason for checking it in this manner.

In this case, you're checking it after it's already been used. 

If you need to confirm it exists, and want to use some JavaScript space wizardry, you could try something like this (comments are for you):

```js
var locale;
// Second part of the condition will only executes if the first part evaluates as true.
// As a bonus, whatever the final part of the expression resolves to gets put in the var
var urlLocale = window.location.search && /locale=([^&]+)&?/.exec(window.location.search);

// If window.location.search is false, null or undefined, or the search returns null,
// !!urlLocale will be false and trigger the default.
locale = !!urlLocale ? urlLocale[1] : (typeof (brackets) !== "undefined" ? brackets.app.language : navigator.language);
```